examples/keras_rs/dlrm.py

```diff
@@ -0,0 +1,480 @@
+    normalization_layers[feature_name] = keras.layers.Normalization(axis=-1)
+
+training_data_for_adaptation = ratings_ds.take(80_000).map(lambda x, y: x)
+
+for feature_name in MOVIELENS_CONFIG["continuous_features"]:
+    feature_ds = training_data_for_adaptation.map(
+        lambda x: tf.expand_dims(x[feature_name], axis=-1)
+    )
+    normalization_layers[feature_name].adapt(feature_ds)
+
+ratings_ds = ratings_ds.map(
+    lambda x, y: (
+        {
+            **{
+                feature_name: lookup_layers[feature_name](x[feature_name])
+                for feature_name in vocabularies
+            },
+            # Apply the adapted normalization layers to the continuous features.
+            **{
+                feature_name: tf.squeeze(
+                    normalization_layers[feature_name](
+                        tf.expand_dims(x[feature_name], axis=-1)
+                    ),
+                    axis=-1,
+                )
+                for feature_name in MOVIELENS_CONFIG["continuous_features"]
+            },
+        },
+        y,
+    )
+)
+
+"""
+Let's split our data into train and test sets. We also use `cache()` and
+`prefetch()` for better performance.
+"""
+
+ratings_ds = ratings_ds.shuffle(100_000)
+
+train_ds = (
+    ratings_ds.take(80_000)
+    .batch(MOVIELENS_CONFIG["batch_size"])
+    .cache()
+    .prefetch(tf.data.AUTOTUNE)
+)
+test_ds = (
+    ratings_ds.skip(80_000)
+    .batch(MOVIELENS_CONFIG["batch_size"])
+    .take(20_000)
+    .cache()
+    .prefetch(tf.data.AUTOTUNE)
+)
+
+"""
+### Building the model
+
+The model will have embedding layers for categorical features, dense layer for continuous
+features followed by DotInteraction and feedforward
+layers.
+"""
+
+
+class DLRM(keras.Model):
+    def __init__(
+        self,
+        dense_num_units_lst,
+        embedding_dim=MOVIELENS_CONFIG["embedding_dim"],
+        mlp_dim=MOVIELENS_CONFIG["mlp_dim"],
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        # Layers for categorical features.
+        self.embedding_layers = {}
+        for feature_name in (
+            MOVIELENS_CONFIG["categorical_int_features"]
+            + MOVIELENS_CONFIG["categorical_str_features"]
+        ):
+            vocab_size = len(vocabularies[feature_name]) + 1  # +1 for OOV token
+            self.embedding_layers[feature_name] = keras.layers.Embedding(
+                input_dim=vocab_size,
+                output_dim=embedding_dim,
+            )
+
+        # Layers for continuous features.
+        self.continuous_dense_layers = {}
+        for feature_name in MOVIELENS_CONFIG["continuous_features"]:
+            # Use a small MLP to process continuous features
+            self.continuous_dense_layers[feature_name] = keras.Sequential(
+                [
+                    keras.layers.Dense(mlp_dim, activation="relu"),
+                    keras.layers.Dense(mlp_dim),
+                ]
+            )
```

Comment:
Generally, we concatenate all dense features together, and use just one MLP. Let's do that, instead of creating separate MLPs from every dense feature.